service_catalog/views/common.py

```diff
@@ -5,26 +5,215 @@
+            color = map_class_to_color.get(map_dict_request_state.get(state))
+            path['datasets'][0]['backgroundColor'].append(color)
+
+        pie_chart_service_tmp = {}
+        queryset = Instance.objects.filter()
+        for instance in queryset:
+            key = instance.service.name
+            pie_chart_service_tmp[key] = pie_chart_service_tmp.get(key, 0) + 1
+        path = context['pie_chart_service']['data']
+        for state, count in pie_chart_service_tmp.items():
+            path['labels'].append(state)
+            path['datasets'][0]['data'].append(count)
+            color = get_color_from_string(path['labels'][-1])
+            color_num += 1
+            path['datasets'][0]['backgroundColor'].append(color)
+
+        pie_chart_billing_tmp = {}
+        queryset = Instance.objects.filter()
+        for instance in queryset:
+            key = instance.billing_group
+            pie_chart_billing_tmp[key] = pie_chart_billing_tmp.get(key, 0) + 1
+        path = context['pie_chart_billing']['data']
+        color_num=0
+        for billing_group, count in pie_chart_billing_tmp.items():
+            path['labels'].append(billing_group.name if billing_group else "None")
+            path['datasets'][0]['data'].append(count)
+            color = get_color_from_string(path['labels'][-1])
+            color_num += 1
+            path['datasets'][0]['backgroundColor'].append(color)
+
+        # Dashboards for resource pool
+
+        chart_resource_pool_tmp = {}
+        # chart_resource_pool_tmp = {
+        #     'vcenter': {
+        #         'vcpu': {
+        #             '5G': 10,
+        #             'orchestration': 30,
+        #         },
+        #         'memory': {
+        #             '5G': 10,
+        #             'orchestration': 30,
+        #         },
+        #
+        #     },
+        #     'ocp': {
+        #         'cpu': {
+        #             '5G': 100,
+        #             'orchestration': 300,
+        #         },
+        #         'ram': {
+        #             '5G': 10,
+        #             'orchestration': 30,
+        #         },
+        #
+        #     }
+        # }
+
+        for resource_pool in ResourcePool.objects.all():
+            chart_resource_pool_tmp[resource_pool] = dict()
+            for rp_attribute in resource_pool.attribute_definitions.all():
+                chart_resource_pool_tmp[resource_pool][rp_attribute] = dict()
+                chart_resource_pool_tmp[resource_pool][rp_attribute][None] = 0
+
+                for bg in BillingGroup.objects.all():
+                    chart_resource_pool_tmp[resource_pool][rp_attribute][bg] = 0
+                for consumer in rp_attribute.consumers.all():
+                    for resource_attribute in consumer.attribute_types.all():
+                        try:
+                            bg = resource_attribute.resource.service_catalog_instance.billing_group
```

Comment:
service_catalog_instance to be renamed 'instance'

Reply:
To much work, it's better to open a new issue. I found a lot of occurrences of service_catalog_instance".